File: advanced/Conditional_Iterative/Password_Validation/pass.py

def password_validation(pwd):
    if len(pwd) < 8:
        sentence = "The password must be longer than 8 charsacters"
    elif len(pwd) >16:
        sentence = "The password must be less than 17 characters"
    dic= {}
    for char in pwd:
        if ("special" not in dic) and char in "+*-/":
            dic["special"] = []
            dic["special"].append(char)
        elif ("special" in dic) and char in "+*-/":
            dic["special"].append(char)
            
        elif ("uppercase" not in dic) and char.isupper():
            dic["uppercase"] = []
            dic["uppercase"].append(char)
        elif ("uppercase" in dic) and char.isupper():
            dic["uppercase"].append(char)
            
        elif ("lowercase" not in dic) and char.islower():
            dic["lowercase"] = []
            dic["lowercase"].append(char)
        elif ("lowercase"  in dic) and char.islower():
            dic["lowercase"].append(char)

        elif ("digit" not in dic) and char.isdigit():
            dic["digit"]= []
            dic["digit"].append(char)
        elif ("digit" in dic) and char.isdigit():
            dic["digit"].append(char)

    if  len(dic["uppercase"])>=2 and len(dic["special"]) >=2 and len(dic["lowercase"])>=2 and len(dic["digit"])>=2:
        sentence= "valid password"
    else:
        sentence = "password must contain 2 characters of a-z, A-Z, 2 digits, and 2 special chars of [+*-/]"
    
    return sentence

    
print(password_validation("Hi2345//*+HGHl"))

File: advanced/Conditional_Iterative/Password_Validation/pass.php
<?php
function passwordValidation($pwd){
    $pwdArray = str_split($pwd);
    $specialchar=0;
    $uppercase=0;
    $lowercase=0;
    $num= 0;
    $extra=0;
    $counter=0;
    foreach ($pwdArray as $char){
        $counter++;
        if ($char== "/" || $char == "-" || $char == "+" || $char == "*"){
            $specialchar++;
        } elseif (ctype_upper($char)){
            $uppercase++;
        } elseif (ctype_lower($char)){
            $lowercase++;
        } elseif (is_numeric($char)){
            $num++;
        } else {
            $extra++;
        }
    }
    if ($counter>=8 && $counter<=16){
        if ($specialchar>1 && $uppercase>1 && $lowercase>1 && $num>1 && $extra==0){
            echo "valid password";
        } else {
            echo "password must contain 2 characters of a-z, A-Z, 2 digits, and 2 special chars of [+*-/]";
        }
    } else {
        echo "password must have btween 8-16 characters";
    }
}

passwordValidation("aBJbG85/-");
?>